server/s3_bucket_metadata.cc

```diff
@@ -279,6 +279,32 @@ std::string S3BucketMetadata::get_tags_as_xml() {
   return tags_as_xml_str;
 }
 
+std::string S3BucketMetadata::get_bucket_versioning_status_as_xml() {
+  s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  std::string versioning_status;
+  std::string versioning_status_as_xml_str;
+
+  if (bucket_versioning_status != "Unversioned") {
+    versioning_status +=
+        "<VersioningConfiguration "
+        "xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\">" +
+        S3CommonUtilities::format_xml_string("Status",
+                                             bucket_versioning_status.c_str()) +
+        "</VersioningConfiguration>";
+
```

Comment:
can we divide api in 
get_bucket_versioning_status
and
get_bucket_versioning_status_as_xml

get api should be used instead of bucket_versioning_status variable

also it needed as other api need to check version status to take decision that time get_bucket_versioning_status api can be used

Reply:
Yes, having separate get_bucket_versioning_status() API would be helpful and we can write one when needed.
As far as get_bucket_versioning_status_as_xml() is concerned, I think we can refer directly to the bucket_versioning_status variable since it is a class method.

Reply:
Resolved

Reply:
Can we have get_bucket_versioning_status() API as part of this PR as this API will be used by GetObject and PutObject to figure out the bucket versioning status?

Reply:
Sure, I will add one.

Reply:
done, added at line #100 above.